app/models/publisher.rb

```diff
@@ -19,6 +19,27 @@ def publish
     end
   end
 
+  def discard_draft
+    begin
+      ActiveRecord::Base.transaction do
+        publishing_api.discard_draft(content_model.content_id)
+
+        latest_published_edition = content_model.editions.published.last
+        if latest_published_edition.present?
+          content_model
+            .editions
+            .where("created_at > ?", latest_published_edition.created_at)
+            .destroy_all
```

Comment:
It'd be nice to remove as much knowledge about the rest of the system as possible here. Making `content_model.lastest_published_edition` (or even `content_model.previously_published?`) and `content_model.editions_since_last_published` kind of thing.